vochain/transaction/account_tx.go

```diff
@@ -239,3 +240,133 @@ func (t *TransactionHandler) SetAccountInfoTxCheck(vtx *vochaintx.Tx) error {
 	}
 	return nil
 }
+
+// DelSikTxCheck checks if a delete sik tx is valid
+func (t *TransactionHandler) DelSikTxCheck(vtx *vochaintx.Tx) (common.Address, error) {
+	if vtx == nil || vtx.Signature == nil || vtx.SignedBody == nil || vtx.Tx == nil {
+		return common.Address{}, ErrNilTx
+	}
+	tx := vtx.Tx.GetDelSik()
+	if tx == nil {
+		return common.Address{}, fmt.Errorf("invalid transaction")
+	}
+
+	pubKey, err := ethereum.PubKeyFromSignature(vtx.SignedBody, vtx.Signature)
+	if err != nil {
+		return common.Address{}, fmt.Errorf("cannot extract public key from vtx.Signature: %w", err)
+	}
+	txAddress, err := ethereum.AddrFromPublicKey(pubKey)
+	if err != nil {
+		return common.Address{}, fmt.Errorf("cannot extract address from public key: %w", err)
+	}
+
+	if _, err := t.state.GetAccount(txAddress, false); err != nil {
+		return common.Address{}, fmt.Errorf("cannot get tx account: %w", err)
+	}
+	return txAddress, nil
+}
+
+// SetSikTxCheck checks if a set sik tx is valid
+func (t *TransactionHandler) SetSikTxCheck(vtx *vochaintx.Tx) (common.Address, vstate.SIK, error) {
+	if vtx == nil || vtx.Signature == nil || vtx.SignedBody == nil || vtx.Tx == nil {
+		return common.Address{}, nil, ErrNilTx
+	}
+	tx := vtx.Tx.GetSetSik()
+	if tx == nil {
+		return common.Address{}, nil, fmt.Errorf("invalid transaction")
+	}
+	pubKey, err := ethereum.PubKeyFromSignature(vtx.SignedBody, vtx.Signature)
+	if err != nil {
+		return common.Address{}, nil, fmt.Errorf("cannot extract public key from vtx.Signature: %w", err)
+	}
+	txAddress, err := ethereum.AddrFromPublicKey(pubKey)
+	if err != nil {
+		return common.Address{}, nil, fmt.Errorf("cannot extract address from public key: %w", err)
+	}
+
+	if _, err := t.state.GetAccount(txAddress, false); err != nil {
+		return common.Address{}, nil, fmt.Errorf("cannot get tx account: %w", err)
+	}
+
+	newSik := vtx.Tx.GetSetSik().GetSik()
+	if newSik == nil {
+		return common.Address{}, nil, fmt.Errorf("no sik value provided")
+	}
+	// check if the address already has invalidated sik to ensure that it is
+	// not updated after reach the correct height to avoid double voting
+	if currentSik, err := t.state.SIKFromAddress(txAddress); err == nil {
+		maxEndBlock, err := t.state.ProcessBlockRegistry.MaxEndBlock(t.state.CurrentHeight(), false)
```

Comment:
Please use app.Height() instead

Reply:
It seems that i have not access to the app here... But probably i am wrong